code/datums/station_traits/negative_traits.dm

```diff
@@ -410,3 +410,254 @@
 					continue
 
 			CHECK_TICK
+
+///Station traits that influence the space background and apply some unique effects!
+/datum/station_trait/nebula
+	name = "Nebula"
+	trait_flags = STATION_TRAIT_ABSTRACT
+	weight = 0
+
+	show_in_report = TRUE
+	can_revert = FALSE //we're too invasive to properly revert. if you feel inspired, im sure you can make it work
+	planetary = FALSE //there's no space so the idea doesnt really work here
+
+	///The color of the "nebula" we send to the players client
+	var/nebula_color
+	///The parallax layer of the nebula
+	var/nebula_layer = /atom/movable/screen/parallax_layer/random/space_gas
+	///The color space 'glows'
+	var/space_light_color = COLOR_STARLIGHT
+	///If set, gives the basic carp different colors
+	var/carp_color_override
+
+/datum/station_trait/nebula/New()
+	. = ..()
+
+	///We set the parallax layer to give a visual effect
+	SSparallax.random_layer = nebula_layer
+	SSparallax.random_parallax_color = nebula_color //give a unique color to tell the player somethings up
+	GLOB.starlight_color = space_light_color //color starlight in our nebula color
+
+	//parallax is generated for quick-joiners before we can change it, so reset it for them :/
+	for(var/client/client as anything in GLOB.clients)
+		client.parallax_layers_cached?.Cut()
+		client.mob?.hud_used?.update_parallax_pref(client.mob)
+
+	//Color the carp in unique colors to better blend with the nebula
+	if(carp_color_override)
+		var/mob/living/basic/carp/carp_type = /mob/living/basic/carp
+		carp_type.carp_colors = carp_color_override
+		//Statics are stored globally, so if we acces a static we never actually acces the object and byond will throw a warning that the var is unused
+		//So we set it to itself to trick byond into working the way it's intended
+		carp_type = carp_type
+
+///Station nebula that incur some sort of effect if no shielding is created
+/datum/station_trait/nebula/hostile
+	trait_flags = STATION_TRAIT_ABSTRACT
+	trait_processes = TRUE
+
+	///Intensity of the nebula
+	VAR_PROTECTED/nebula_intensity = -1
+	///The max intensity of a nebula
+	var/maximum_nebula_intensity = 2 HOURS
+	///How long it takes to go to the next nebula level/intensity
+	var/intensity_increment_time = INFINITE
```

Comment:
Why are these not protected but nebula_intensity is? From a quick ctrl+f looks like only subtypes use it